File: modules/widgets/lib/rin/widgets/event/ScrollEvent.cpp
#include "rin/widgets/event/ScrollEvent.hpp"

ScrollEvent::ScrollEvent(const Shared<WidgetSurface>& inSurface, const Vec2<float>& inDelta,
                         const Vec2<float>& inPosition) : Event(inSurface), delta(inDelta), position(inPosition)
{
}

File: rin.Widgets/Events/ScrollEvent.cs
using rin.Core.Math;
using rin.Widgets.Graphics;

namespace rin.Widgets.Events;

public class ScrollEvent(Surface surface, Vector2<float> position, Vector2<float> delta)
    : Event(surface)
{
    public Vector2<float> Delta = delta;
    public Vector2<float> Position = position;
}